library/manageiq_user.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/python
+
+
+DOCUMENTATION = '''
+---
+module: manageiq_user
+short_description: management of users in ManageIQ
+requirements: [ ManageIQ/manageiq-api-client-python ]
+author: Daniel Korn (@dkorn)
+options:
+  miq_url:
+    description:
+      - the manageiq environment url
+    default: MIQ_URL env var if set. otherwise, it is required to pass it
+  miq_username:
+    description:
+      - manageiq username
+    default: MIQ_USERNAME env var if set. otherwise, it is required to pass it
+  miq_password:
+    description:
+      - manageiq password
+    default: MIQ_PASSWORD env var if set. otherwise, it is required to pass it
+  userid:
+    description:
+      - the unique userid in manageiq, often mentioned as username
+    required: true
+    default: null
+  username:
+    description:
+      - the users' full name
```

Comment:
So `userid` is "unique id, often mentioned as username" and `username` is "full name"?
AFAICT idempotency here identifies by `userid`, updates `username` if differs.
I think there is a strong tradition the word "username" can only mean the unique identifying string ("dkorn").
There is also some tradition that user "id" is numeric.

=> I suggest renaming this field from `username` to `full_name` (or `user_full_name`), and maybe renaming `userid` to `username`.